hydradx/model/amm/stableswap_amm.py

```diff
@@ -230,11 +230,10 @@ def swap(
         elif self.liquidity[tkn_buy] <= buy_quantity:
             return self.fail_transaction('Pool has insufficient liquidity.')
 
-        new_agent = agent  # .copy()
-        if tkn_buy not in new_agent.holdings:
-            new_agent.holdings[tkn_buy] = 0
-        new_agent.holdings[tkn_buy] += buy_quantity
-        new_agent.holdings[tkn_sell] -= sell_quantity
+        if tkn_buy not in agent.holdings:
```

Comment:
This seems like a critical change, was this a bug we were doing analyses with up until now?